dashboard/app/models/pd/workshop.rb

```diff
@@ -269,4 +292,36 @@ def self.process_location(address)
       formatted_address: result.formatted_address
     }.to_json
   end
+
+  # Min number of days a teacher must attend for it to count
+  def min_attendance_days
+    constraints = TIME_CONSTRAINTS_BY_SUBJECT[self.subject]
+    if constraints
+      constraints[:min_days]
+    else
+      1
+    end
+  end
+
+  # Apply max # days for payment, if applicable, to the number of scheduled days (sessions).
+  def effective_num_days
+    max_days = TIME_CONSTRAINTS_BY_SUBJECT[self.subject].try{|constraints| constraints[:max_days]}
+
+    if max_days
+      [self.sessions.count, max_days].min
+    else
+      self.sessions.count
+    end
```

Comment:
This could be simplified to `[self.sessions.count, max_days].compact.min`? Similarly in `effective_num_hours` below.